acvm-repo/acvm/src/compiler/optimizers/merge_expressions.rs

```diff
@@ -27,9 +27,25 @@ impl<F: AcirField> MergeExpressionsOptimizer<F> {
         }
     }
     /// This pass analyzes the circuit and identifies intermediate variables that are
-    /// only used in two gates. It then merges the gate that produces the
+    /// only used in two arithmetic opcodes. It then merges the opcode which produces the
     /// intermediate variable into the second one that uses it
     /// Note: This pass is only relevant for backends that can handle unlimited width
+    ///
+    /// The first pass maps witness with the opcodes using it.
+    /// Public inputs are not considered because they cannot be simplified.
+    /// Witnesses used by MemoryInit opcodes are put in a separate map and marked as used by a Brillig call
+    /// if the memory block is an input to the call.
+    ///
+    /// The second pass looks for arithmetic opcodes having a witness which is only used by another arithmetic opcode.
+    /// In that case, the opcode with the smallest index is merged into the other one via Gaussian elimination.
+    /// For instance, if we have '_1' used only by these two opcodes:
```

Comment:
```suggestion
    /// Where `_{value}` refers to a witness and `{value}` refers to a constant, 
    /// if we have '_1' used only by these two opcodes:
```
Would be good to clarify `_1` is a witness. Someone reading this would have to infer this is the witness or would have to be familiar with how we print ACIR. Clarifying in this way also let's us prevent the comment going out of date if we change how we print ACIR but do not update this comment.

Reply:
Done